src/context/PortfolioProvider/PortfolioContext.tsx

```diff
@@ -174,6 +178,44 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
     })()
   }, [assetsById, chainAdapter, dispatch, wallet])
 
+  const txIds = useSelector(selectTxIds)
+  const txsById = useSelector(selectTxs)
+  const txHistoryStatus = useSelector(selectTxHistoryStatus)
+
+  /**
+   * portfolio refetch on new tx logic
+   */
+  useEffect(() => {
+    // we only want to refetch portfolio if a new tx comes in after we're finished loading
+    if (txHistoryStatus !== 'loaded') return
+    // don't fire with nothing connected
+    if (isEmpty(accountSpecifiers)) return
+    // grab the most recent txId
+    const txId = head(txIds)!
+    // grab the actual tx
+    const tx = txsById[txId]
+    // always wear protection, or don't it's your choice really
+    if (!tx) return
+    // the chain the tx came from
+    const txChainId = tx.caip2
+    dispatch(portfolio.actions.clearAssetsBalanceByChain({ chain: txChainId }))
```

Comment:
this is kinda the right idea, but will cause a flicker in the UI, as we're going to zero out the balances, they'll render lower, then come back higher/lower when the getAccount call comes back

i'd rather either dispatch this when we have the new data ready to upsert in the getAccount api, or upsert correctly as to only cause one render